packages/react/src/hooks/useIsClient.ts

```diff
@@ -1,12 +1,7 @@
-import { useState } from 'react'
-import { useIsomorphicLayoutEffect } from './useIsomorphicLayoutEffect'
+import { useSyncExternalStore } from 'react'
+import { noop } from '../utils'
 
-export const useIsClient = () => {
-  const [isClient, setIsClient] = useState(false)
-
-  useIsomorphicLayoutEffect(() => {
-    setIsClient(true)
-  }, [])
-
-  return isClient
-}
+const emptySubscribe = () => noop
+const getSnapshot = () => true
+const getServerSnapshot = () => false
+export const useIsClient = () => useSyncExternalStore(emptySubscribe, getSnapshot, getServerSnapshot)
```

Comment:
@TkDodo If you okay, I want to request you to review this useIsClient hook.
When I test this in Next.js, I have no issue too. I also saw [this tweet](https://x.com/TkDodo/status/1741068994981826947?t=XmG17etMUL2m0JFim03vqw&s=19) and [this article](https://tkdodo.eu/blog/avoiding-hydration-mismatches-with-use-sync-external-store) but since this is the interface you proposed, I'd really like to hear your thoughts. This improvement could be big impact for even our all product of my company(@toss) and all users of Suspensive libraries

ps. @gwansikk and I are preparing @suspensive/react-query everyday to support @tanstack/react-query as one of community resources. (Recently [we start to support not only TanStack Query v4, also v5 at once](https://suspensive.org/docs/react-query/tanstack-query-compatibility) by using postinstall script like [vue-demi](https://github.com/vueuse/vue-demi) used by @tanstack/vue-query too.) I'm learning a lot from you and TanStack Query and I rely on TanStack Query v4, v5 for all my projects. Thank you!